core/arxiv/submission/rules/set_title.py

```diff
@@ -0,0 +1,154 @@
+                            creator=creator,
+                            matching_id=ident,
+                            matching_title=title,
+                            matching_owner=submitter))
+           for ident, title, submitter in candidates
+           if above_similarity_threshold(jaccard(event.title, title)))
+    return chain(remove, add)
+
+
+@SetTitle.bind(condition=lambda *a: not system_event(*a))
+def check_for_excessive_unicode(event: SetTitle, before: Submission,
+                                after: Submission, creator: Agent) \
+        -> Iterable[Event]:
+    """
+    Screen for possible abuse of unicode in titles.
+
+    We support unicode characters in titles, but this can get out of hand.
+    This rule adds an annotation if the ratio of non-ASCII to ASCII characters
+    is too high.
+    """
+    if below_ascii_threshold(proportion_ascii(event.title)):
+        return [AddAnnotation(
+            creator=creator,
+            annotation=PossibleMetadataProblem(
+                creator=creator,
+                field_name='title',
+                description='Possible excessive use of non-ASCII characters.'
+            )
+        )]
+    return []
+
+
+def proportion_ascii(phrase: str) -> float:
+    """Calculate the proportion of a string comprised of ASCII characters."""
+    return len([c for c in phrase if is_ascii(c)])/len(phrase)
+
+
+@lru_cache(maxsize=1028)    # Memoized
+def normalize(phrase: str) -> str:
+    """Prepare a phrase for tokenization."""
+    return unidecode(phrase.lower()).translate(REMOVE_PUNCTUATION)
+
+
+@lru_cache(maxsize=2056)    # Memoized
+def tokenized(phrase: str) -> Set[str]:
+    """Split a phrase into tokens and remove stopwords."""
+    return set(normalize(phrase).split()) - STOPWORDS
+
+
+def intersection(phrase_a: str, phrase_b: str) -> int:
+    """The number tokens shared by two phrases."""
+    return len(tokenized(phrase_a) & tokenized(phrase_b))
+
+
+def union(phrase_a: str, phrase_b: str) -> int:
+    """The total number tokens in two phrases."""
+    return len(tokenized(phrase_a) | tokenized(phrase_b))
+
+
+def jaccard(phrase_a: str, phrase_b: str) -> float:
+    """The Jaccard similarity of two phrases."""
+    return intersection(phrase_a, phrase_b) / union(phrase_a, phrase_b)
+
+
+def above_similarity_threshold(similarity: float) -> bool:
+    """Whether or not a Jaccard similarity is above the threshold."""
+    threshold = get_application_config().get('TITLE_SIMILARITY_THRESHOLD', 0.7)
+    return similarity > threshold
+
+
+def below_ascii_threshold(proportion: float) -> bool:
+    """Whether or not the proportion of ASCII characters is too low."""
+    threshold = get_application_config().get('TITLE_ASCII_THRESHOLD', 0.5)
+    return proportion < threshold
+
+
+def window() -> datetime:
+    """Get the time window for possible duplicate submissions."""
+    days = get_application_config().get('TITLE_SIMILARITY_WINDOW', 3*365/12)
```

Comment:
Why not just use `365 / 4`?

Reply:
Takes slightly less cognitive effort if we want to change it to 2 months or 4 months ;-)

Reply:
"months"